content/03.text-mining.md

```diff
@@ -21,34 +21,67 @@
+In the rest of this section we discuss approaches that use supervised methods to detect relationship asserting sentences.
+
+Extracting relationships in a supervised setting requires the capacity to map textual input onto a high dimensional space.
+Support vector machines are a type of classifier that can accomplish this mapping problem by utilizing a mapping function called a kernel [@doi:10.1186/s13326-017-0168-3; @doi:10.1371/journal.pcbi.1004630].
+These kernels take information such as a sentence's dependency tree [@doi:10.1093/database/bay108; @doi:10.1093/bioinformatics/btw503], part of speech tags [@doi:10.1186/s13326-017-0168-3] or even word counts [@doi:10.1371/journal.pcbi.1004630] and directly maps them onto a dense feature space.
+Within this space a hyperplane can be constructed to seperate sentences into the positive class (mentions a relationship) or negative class (does not mention a relationship). 
+Constructing these kernels necessitates the use of expert knowledge to be successful.
+Similar to regular manual curation, relying on experts to manually construct functions result in highly precise performance, but suffer overall in terms of recall.
+Besides single kernel methods, a recent study used an ensemble of SVMs to extract disease-gene associations [@doi:10.1371/journal.pone.0200699].
+This provided powerful preformance in terms of precision and outperformed the various baseline models.
+Overall, future SVM methods should consider a more automated approach to construct these kernels or develop ways to use semi-automatic methods for kenel construction.
+
+Deep learning is an increasingly popular field that can construct their own features within a high dimensional space [@raw:GoodfellowDL; @doi:10.1038/nature14539].
+These methods use different forms of neural networks, such a recurrent neural network or a convolutional neural network, to perform classifcation.
+Recurrent neural networks (RNN) are designed for sequential analysis that consist of using a repeatedly updating hidden state to make predictions.
+An example of a recurrent neural network is a long short term memory (LSTM) network [@doi:10.1162/neco.1997.9.8.1735].
+Cocos et al [@doi:10.1093/jamia/ocw180] used a LSTM to extract drug side effects from de-identified twitter posts, while Yadav et al [] used an LSTM to extract protein protein interactions.
+Other works have used LSTMs to perform relationship extraction [@arxiv:1708.03743; @doi:10.1093/bioinformatics/btw486; @arxiv:1808.09101; @doi:10.1186/s12859-017-1609-9; @doi:10.1093/jamia/ocw180]. 
+Despite the success of these networks, runtime for RNNs can be an issue.
+Using a LSTM on a dataset that has 200+ token sentences can take up to days to run.
+Based on this future approaches should be cautious on length of each individual sentence using this type of network.
```

Comment:
I think you mean people using these approaches should be cautious - not the approaches themselves. Are there any refs for good ways to deal with this? Drop long sentences? Something else?